index.js

```diff
@@ -1,97 +1,101 @@
+  var extensionsDir = extensionDir(profileDir);
+  var directories = listDirectories(extensionsDir);
+  return directories;
+}
 
-  var b2g_profiles = [];
 
-  fs.readdir(FX_PROFILES, function (err, profiles) {
+function listDirectories(directoryPath) {
+  var dirContents = fs.readdirSync(directoryPath);
 
-    // Filter profiles that don't have extensions
-    profiles = profiles.filter(extensionsExist);
+  return dirContents.map(function(entry) {
+    return path.join(directoryPath, entry);
+  }).filter(isDir);
+}
 
-    // Iterate through all the profiles to find extensions
-    async.each(profiles, function (profile, next) {
 
-      var extensions_path = path.join(FX_PROFILES, profile, 'extensions');
+function flatten(arrays) {
+  if(arrays.length === 0) {
+    return [];
+  } else {
+    var firstElement = arrays[0];
+    var restOfArrays = arrays.slice(1);
+    return firstElement.concat(flatten(restOfArrays));
+  }
+}
 
-      // Iterate in each extension to find simulators
-      fs.readdir(extensions_path, function(err, extensions) {
-        
-        async.each(extensions, function(extension, done) {
-        
-          var matches = matchSimulator(extension);
-          if (matches && matches[1]) {
-            var release = matches[1].replace('_','.');
 
-            // if we want a specific release, skip it
-            if (opts.release && opts.release.indexOf(release) < 0)
-              return done(null);
+function getSimulatorInfo(extensionDir) {
+  var simulatorRegex = /fxos_(.*)_simulator@mozilla.org$/;
+  var matches = simulatorRegex.exec(extensionDir);
+  if(matches && matches[1]) {
+    var version = matches[1].replace('_', '.');
+    return {
+      version: version,
+      bin: path.join(extensionDir, currentPlatform.simulatorBinary),
+      profile: path.join(extensionDir, 'profile')
+    };
+  } else {
+    return null;
+  }
+}
 
-            b2g_profiles.push({
-              release: release,
-              bin: path.join(extensions_path, extension, B2G_BIN),
-              profile: path.join(extensions_path, extension, 'profile')
-            });
-          }
 
-          done(null);
-        }, next);
+function findSimulators(options) {
 
-      });
+  options = options || {};
 
-    }, function(err) {
-      callback(err, b2g_profiles);
-    });
+  var simulators = [];
+  var profilesDir = currentPlatform.firefoxProfilesDir;
+  var profileFolders = listDirectories(profilesDir);
+  var profilesWithExtensions = profileFolders.filter(hasExtensions);
 
+  // per profile -> get extensions, then filter to only simulator extensions
+  var extensionsPerProfile = profilesWithExtensions.map(listExtensions);
+  var extensions = flatten(extensionsPerProfile);
+  var extensionsWithInfo = extensions.map(getSimulatorInfo);
+  simulators = extensionsWithInfo.filter(function(info) {
+    return info !== null;
   });
 
+  if(options.version) {
+    var version = options.version;
+    if(version instanceof String) {
```

Comment:
The linter will catch these if you wanna wait, but I just want to cement my status as humanoid style checker ;-)